### ISSUE TITLE
bug: local jobs start with a very long delay on FreeBSD

### ISSUE THREAD

**Issue by yurivict** [pmodels/mpich#3315]
I run a simple Hello World job printing only one line (http://mpitutorial.com/tutorials/mpi-hello-world/).
There is a very long delay before jobs run:
```
$ time mpirun -np 20 hw
Hello world from processor myhost, rank 1 out of 20 processors
Hello world from processor myhost, rank 5 out of 20 processors
...skipped similar lines....
Hello world from processor myhost, rank 19 out of 20 processors

real	0m13.768s
user	0m0.097s
sys	0m0.008s
```

20 jobs were delayed for 13.5+ seconds before all of them ran at about the same time.

mpd was started with ```mpdboot -n 1 --ncpus=8``` with only the IP of the local host with 8 cpus in ```mpd.hosts```.

mpich2-1.5_11,5 on FreeBSD 11.2

**Comment by raffenet** [comment]
Are you able to use a newer version of MPICH? It would be useful to know if the problem persists in MPICH 3.2.1.

**Comment by hzhou** [comment]
The original issue is reported against `mpd`, which is deprecated since 3.1.1. However, the similar symptom is still being experienced -- due to different parts of code but probably due to a similar cause -- interface name to ip look up.

For example, a user reported on the discussion list today:
```
I use OpenVPN system on my pc, and while I want to use the mpiexec it run too slowly ( e.g. takes 1min to run hello world), but when I disable the ethernet connection that corresponds to the Openvpn, mpiexec works promptly. The problem is that I cant use both my OpenVPN and mpiexec at the same time every time I have to disable it. How can I be able to stop the mpiexec to connect to the OpenVpn?
```

I also experience this on my laptop: 
```
$ mpiexec -np 1 out/test_send
Fatal error in MPI_Init: Other MPI error, error stack:
MPIR_Init_thread(565)..............: 
MPID_Init(224).....................: channel initialization failed
MPIDI_CH3_Init(105)................: 
MPID_nem_init(324).................: 
MPID_nem_tcp_init(178).............: 
MPID_nem_tcp_get_business_card(425): 
MPID_nem_tcp_init(384).............: gethostbyname failed, Huis-MacBook-Pro.local (errno 1)
```
```
$ time mpiexec -np 2 -host localhost out/test_send
Hello world from Huis-MacBook-Pro.local, 0/2
Hello world from Huis-MacBook-Pro.local, 1/2
Node 1: received data from node 0: 807

real    0m5.078s
user    0m0.043s
sys     0m0.036s
```

**Comment by hzhou** [comment]
The long delay is in `mpiexec.hydra`, or precisely, in the call to `MPL_host_is_local(node->hostname)`.

**Comment by hzhou** [comment]
The proper fix won't go in for a while (as #3363 wasn't on any list yet), and I yet again had this issue in a hotel :(. Problem chokes for same issue but at different places depend on whether the laptop is connected to some network or not. Here is a temporary patch I needed (for the hotel case):
[`mpl_host.c` -`init_lhost_list()` - right after `append_lhost(tmp_lhost);` and before `gethostbyname`]
insert:
```
    int n=strlen(tmp_lhost);
    if (n>6 && strcmp(tmp_lhost+n-6, ".local")==0){
        goto common_names;
    }
```

**Comment by hzhou** [comment]
We also need patch `src/mpid/ch3/channels/nemesis/netmod/tcp/tcp_init.c`, line 355:
```
+    int n=strlen(ifname_string);
+    if (n>6 && strcmp(ifname_string+n-6, ".local")==0){
+        ifname_string = "localhost";
+    }
```
